src/distribution.cpp

```diff
@@ -287,6 +287,48 @@ double Equiprobable::sample(uint64_t* seed) const
   return xl + ((n - 1) * r - i) * (xr - xl);
 }
 
+//==============================================================================
+// Mixture implementation
+//==============================================================================
+
+Mixture::Mixture(pugi::xml_node node)
+{
+  double cumsum = 0.0;
+  for (pugi::xml_node pair : node.children("pair")) {
+    // Check that required data exists
+    if (!pair.attribute("probability")) fatal_error("Mixture pair element does not have probability.");
+    if (!pair.child("dist")) fatal_error("Mixture pair element does not have a distribution.");
+
+    // cummulative sum of probybilities
+    cumsum += std::stod(pair.attribute("probability").value());
+
+    // Save cummulative probybility and distrubution
+    distribution_.push_back(
+      std::make_pair(cumsum, distribution_from_xml(pair.child("dist"))));
+  }
+
+  // Normalize cummulative probabilities to 1
+  for (auto& pair : distribution_) {
+    pair.first /= cumsum;
+  }
+}
+
+double Mixture::sample(uint64_t* seed) const
+{
+  // Sample value of CDF
+  const double p = prn(seed);
+
+  // find matching distribution
+  const auto it = std::lower_bound(distribution_.cbegin(), distribution_.cend(),
+    p, [](const DistPair& pair, double p) { return pair.first < p; });
+
+  // This should not happen. Catch it
+  Ensures(it != distribution_.cend());
```

Comment:
To get this to work, you'll need to add:
```C++
#include <gsl/gsl-lite.hpp>
```
at the top of the file with other includes.